lib/default.ml

```diff
@@ -24,6 +24,12 @@ let rec expr_from_core_type ~loc {ptyp_desc; ptyp_loc; _} =
   | Ptyp_constr ({txt = Lident "option"; _}, _) -> Ok [%expr None]
   | Ptyp_constr ({txt = Lident "list"; _}, _) -> Ok [%expr []]
   | Ptyp_constr ({txt = Lident "array"; _}, _) -> Ok [%expr [||]]
+  | Ptyp_constr ({txt = Lident "result"; _}, [ok; error]) ->
+    let open Util.Result_ in
+    ( match expr_from_core_type ~loc ok with
+      | Ok ok_arg -> Ok [%expr Ok [%e ok_arg]]
+      | Error _ -> expr_from_core_type ~loc error >|= fun err_arg -> [%expr Error [%e err_arg]]
```

Comment:
I think I'm misunderstanding, but why do we call` expr_from_core_type` again here?

Reply:
No you're right, it seems like a mistake, I'll add a test to make sure and fix it!

Reply:
Ok no my bad it is indeed right and tested.

What happens here is for any `('a, 'b) result` type, we check whether we can derive a default for `'a` in which case the default value is `Ok default_a`. If we can't we try to derive a default for `'b`. If we succeed, the default value if `Error default_b`, otherwise that means we can't derive a default value for the whole result type and we fail.

I changed the `ok` and `error` from above into `ok_type` and `error_type`, let me know if you think that makes things clearer!

Reply:
Yeah that is clearer thanks